### PR TITLE
bundle: Use the standard key id in the JWT header.

### PR DESCRIPTION
**Pull request by koponen-styra**
Instead of having a custom one in the claims.

Signed-off-by: Teemu Koponen <koponen@styra.com>

### CODE REVIEW

File: bundle/verify.go [hunk 1]
```diff
@@ -31,80 +34,85 @@ func VerifyBundleSignature(sc SignaturesConfig, bvc *VerificationConfig) (map[st
...
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	// verify the scope
 	scope := bvc.Scope
 	if scope == "" {
-		scope = keyConfig.Scope
+		//scope = keyConfig.Scope
```
Comment (ashutosh-narkar) [inline bundle/verify.go:109]: Nit: Uncomment

File: docs/content/management.md [hunk 2]
```diff
@@ -341,13 +340,13 @@ The JWT when decoded has a JSON payload of the following form:
...
 | `files[_].algorithm` | `string` | Yes | Name of the hashing algorithm. |
-| `keyid` | `string` | No | Name of the key to use for JWT signature verification. |
+| `kid` | `string` | No | Name of the key to use for JWT signature verification (this is the JWT header, not payload). |
 | `scope` | `string` | No | Represents the fragment of signings. |
 | `iat` | `string` | No | Time of signature creation since epoch in seconds. For informational purposes only. |
 | `iss` | `string` | No | Identifies the issuer of the JWT. For informational purposes only. |
 
 > Note: OPA will first look for the `keyid` on the command-line. If the `keyid` is empty, OPA will look for it in it's
-> configuration. If `keyid` is still empty, OPA will finally look for it in the JWT payload.
+> configuration. If `keyid` is still empty, OPA will finally look for it in the JWT header.
```
[the remark this answers is not in the document]
Comment (koponen-styra) [reply docs/content/management.md:350]: Sure, added more details.

File: bundle/verify_test.go [hunk 3]
```diff
@@ -107,58 +106,80 @@ MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQDMYfnvWtC8Id5bPKae5yXSxQTt
...
+		XEN0BzxCplU/eFPL8bvi03fjnCOh2iG599d8vtOvAoGBAMlB1ZRDLDSMydE2N2+U
+		Bm3qjKyvTU+aLi4ek+rBopJbahrjfp61weg+R4mOoGznGmTu9TWxqjo5+JZTdhAc
+		D5ZUIF5OXT3K+kvaJmVevvOsrpiNl0W451peCZwysFhGv4urRAAV9zumxwc4IndB
+		Z5P5F8COKdj6wvqiXubAuwudAoGAD6gPaLB3DbM35/fXO6JyDhQz3F29plSZ5p1O
+		kt382NPCx4ueAmLIWiWes5KZoMRZl1jMfHQMfsn2SRYrEGDN9rnieYCKk3WNdlHE
+		95k8OmhLt/0rkCulw0V8yR4E+6ZkG6PVm8WrID7t78dWlZ8KCHfsFlm6+tm21SbN
+		jD44t6kCgYEAnTHH8SOMoW/kMckESubV2y4rvdLol7nVX3ia7IGP8TA6NHsGT0Qb
+		EOuKwnuQ3ZToMdNVrS2nFyLe/HohSbT0SrNu2j4YkZDxHGQImZvw/KtMFPyT+Ff8
+		rtvSvuiJNjLUr6OcqxARXrmispBO/GxvWucF0tcpOcSGUDYHAD2yIAQ=
+		-----END RSA PRIVATE KEY-----`*/
```
Comment (patrick-east) [inline bundle/verify_test.go:135]: nit: Remove commented out code or add comment explaining what it is

File: docs/content/management.md [hunk 4]
```diff
@@ -341,13 +341,13 @@ The JWT when decoded has a JSON payload of the following form:
 | `files[_].name` | `string` | Yes | Path of a file in the bundle. |
 | `files[_].hash` | `string` | Yes | Output of the hashing algorithm applied to the file. |
 | `files[_].algorithm` | `string` | Yes | Name of the hashing algorithm. |
-| `keyid` | `string` | No | Name of the key to use for JWT signature verification. |
+| `kid` | `string` | No | Name of the key to use for JWT signature verification (this is in the JWT header, not in payload). |
```
Comment (patrick-east) [inline docs/content/management.md:344]: Should we remove this line from the table? This is describing the payload, right?

**Comment by koponen-styra** [comment]
> 🤔 Should we support the key id being in the payload or header to maintain compatibility? Not sure what kind of guarantees we provide around that or maybe its new enough that it doesn't matter?

Hmm, I'd say this is new enough not to matter.

File: bundle/verify_test.go (hunk 3 above)
Comment (koponen-styra) [reply bundle/verify_test.go]: Added a comment since the compiler would complain about unused variable.

File: docs/content/management.md (hunk 4 above)
Comment (koponen-styra) [reply docs/content/management.md]: Removed.

**Comment by tsandall** [comment]
Looks like the last outstanding issue is this question around backwards compatibility. @koponen-styra would it be possible to support both for the time being (IOW, how much effort would it be)? We can effectively deprecate the keyid claim in the payload (removing it from the docs should be sufficient). This way we don't break anyone who's adopted v0.22 and already started using signing.

**Comment by koponen-styra** [comment]
> Looks like the last outstanding issue is this question around backwards compatibility. @koponen-styra would it be possible to support both for the time being (IOW, how much effort would it be)? We can effectively deprecate the keyid claim in the payload (removing it from the docs should be sufficient). This way we don't break anyone who's adopted v0.22 and already started using signing.

Shouldn't be much. Will update the PR.

**Pull request merged**